drivers/libaio-backend.c

```diff
@@ -110,7 +110,7 @@ struct tio {
 static inline void
 queue_tiocb(libaio_queue *queue, struct tiocb *tiocb)
 {
-	struct iocb *iocb = (struct iocb*)tiocb->iocb;
+	struct iocb *iocb = (struct iocb*)&(tiocb->uiocb);
```

Comment:
This should be `(struct iocb*)&(tiocb->uiocb.io)`, similarly where the aiocb is referenced `(struct aiocb*)&(tiocb->uiocb.aiocb)`. In pointer terms it almost certainly equates to the same thing, both members of the union being aling with the start of the union but it's not correct.

Reply:
You can remove the cast actually in this case, since the union member should have the correct type already (prevents accidentally casting it to a wrong type).

Reply:
Seen many unions in code but actually used one anger, so i've learned a lot here. I will fix this up and retest

Reply:
Fix in latest commit